src/languages/elm.ts

```diff
@@ -0,0 +1,55 @@
+import { SimpleScopeTypeType } from "../typings/targetDescriptor.types";
+import { NodeMatcherAlternative } from "../typings/Types";
+import { createPatternMatchers } from "../util/nodeMatchers";
+
+const nodeMatchers: Partial<
+  Record<SimpleScopeTypeType, NodeMatcherAlternative>
+> = {
+  map: ["record_expr", "record_pattern"],
+  list: ["list_expr", "list_pattern"],
+  statement: [
+    // not all of these are "statements" in Elm, but they fit the usage pattern in Cursorless
+    "let_in_expr",
+    "case_of_expr",
+    "port_annotation",
+    "infix_declaration",
+    "if_else_expr",
+  ],
+  string: "string_constant_expr",
+  collectionItem: ["exposed_value"],
+  collectionKey: ["field_type[name]", "record_expr.field![name]"],
+  ifStatement: "if_else_expr",
+  anonymousFunction: "anonymous_function_expr",
+  functionCall: "function_call_expr",
+  functionCallee: "function_call_expr[target][name]",
+  comment: ["block_comment", "line_comment"],
+  namedFunction: "value_declaration",
```

Comment:
Is this not too generic?

Reply:
Not sure, but tree-sitter-elm does not distinguish functions and other values (see image).
I kinda got used to thinking that values are functions with zero arguments.

I suppose it's not right if you should be able to select this with `take func` and with the cursor within `z = 0`:
`f x = let z = 0 in x`

So I could add `"value_declaration.anonymous_function_expr"` to cover `g = \x -> x`, and then something to select "value_declaration with at least one pattern in function_declaration_left".

<img width="969" alt="image" src="https://user-images.githubusercontent.com/25511262/180789819-f36c3a0c-d02f-4719-85f0-756243b11d0b.png">

Reply:
Huh. Yeah I guess that makes sense?

Feel free to drop into one of the meet-ups if you want to discuss; see [the calendar](https://tinyurl.com/talon-events)